source/common/http/conn_manager_impl.cc

```diff
@@ -1411,7 +1411,7 @@ bool ConnectionManagerImpl::ActiveStream::createFilterChain() {
   state_.created_filter_chain_ = true;
   if (upgrade != nullptr) {
     const Router::RouteEntry::UpgradeMap* upgrade_map = nullptr;
-    if (cached_route_.value() && cached_route_.value()->routeEntry()) {
+    if (cached_route_.has_value() && cached_route_.value() && cached_route_.value()->routeEntry()) {
```

Comment:
Can you add a small comment above here on why we need to check if the optional is populated? It's not immediately obvious (basically that this can get called during early local reply, etc.).